src/adapters/world-name-permission-checker.ts

```diff
@@ -8,11 +8,34 @@ type NamesResponse = {
   nfts: { name: string; owner: { id: string } }[]
 }
 
-export const createDclNameChecker = (
+export async function createWorldNamePermissionChecker(
+  components: Pick<AppComponents, 'config' | 'ethereumProvider' | 'fetch' | 'logs' | 'marketplaceSubGraph'>
+): Promise<IWorldNamePermissionChecker> {
+  const logger = components.logs.getLogger('check-permissions')
+  const nameValidatorStrategy = await components.config.requireString('NAME_VALIDATOR')
+  switch (nameValidatorStrategy) {
+    case 'THE_GRAPH_DCL_NAME_CHECKER':
+      logger.info('Using TheGraph DclNameChecker')
+      return createTheGraphDclNameChecker(components)
+    case 'ON_CHAIN_DCL_NAME_CHECKER':
+      logger.info('Using OnChain DclNameChecker')
+      return await createOnChainDclNameChecker(components)
```

Comment:
Question, on-chain is linked to DCL because it uses our contracts, right?

Reply:
Both on-chain and the-graph implementations are bound to DCL names.